coordinator/internal/transitengine/crypto.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package transitengine
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/edgelesssys/contrast/internal/crypto"
+
+	b64 "encoding/base64"
+)
+
+// symmetricOpts holds parameters related to the performed symmetric encryption, specifyable as http request parameters.
+type symmetricOpts struct {
+	// Convergent TODO(jmxzo): add parameter support
+	convergent bool
+	// Nonce
+	nonce []byte
+	// AdditionalData
+	additionalData []byte
+}
+
+type encryptionMap struct {
+	Plaintext     string `json:"plaintext"`
+	Ciphertext    string `json:"ciphertext"`
+	symmetricOpts symmetricOpts
+}
+
+// Note: In case only err is returned the resulting httpError will fallbback to InternalServerError status code 500.
```

Comment:
Could you elaborate a bit more? Since this is not a function documentation, this floats a bit between the functions/code? Is is related to the function below?